NullPointer when generating after change on 2019-02-23

We have been using the 3.0.0 branch of swagger-codegen to generate som code for our rest APIs. It was working fine until I took a pull of swagger-codegen. I have tracked the issue down to a change made on the 23 of February 2019 in this project (swagger-codegen-generators):
"added contenttype extension for response schema. Hugo Mercado 2019-02-23 20:00"

There was a change then in the file DefaultCodegenConfig on rows: 3755-3757.

Now when I use for instance this yaml to generate:

```
openapi: 3.0.1
info:
  title: Authentication API
  description: Authentication API
  contact:
    url: https://example.net
  license:
    name: Proprietary
    url: https://example.net
  version: "1.0"
servers:
- url: https://example.net
tags:
- name: /authentication
paths:
  /authentication/login_with_token:
    post:
      tags:
      - /authentication
      operationId: loginWithToken
      parameters: []
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/AuthenticationTokenRequest'
        required: true
      responses:
        default:
          content:
            application/json: {}
components:
  schemas:
    AuthenticationTokenRequest:
      type: object
      properties:
        identityToken:
          minLength: 1
          type: string
```

I get a NullPointerException in Default.CodegenConfig.getSchemaFromResponse. Heres part of the log with the error:

```
12:49:57.957 [Thread-1] DEBUG io.swagger.codegen.v3.config.CodegenConfigurator - getClientOptInput - parsed inputSpecURL ../authentication-api-1.0.yml
12:49:58.146 [Thread-1] DEBUG com.github.jknack.handlebars.internal.HbsParserFactory - About to parse: /handlebars/cpp/model-header.mustache
12:49:58.199 [Thread-1] DEBUG com.github.jknack.handlebars.internal.HbsParserFactory - Building AST
12:49:58.241 [Thread-1] DEBUG com.github.jknack.handlebars.internal.HbsParserFactory - Applying Mustache spec
12:49:58.250 [Thread-1] DEBUG com.github.jknack.handlebars.internal.HbsParserFactory - Creating templates
12:49:58.277 [Thread-1] INFO io.swagger.codegen.v3.AbstractGenerator - writing file /var/tmp/authentication-client/example/api/authentication/AuthenticationTokenRequest.h
java.lang.NullPointerException
        at io.swagger.codegen.v3.generators.DefaultCodegenConfig.getSchemaFromResponse(DefaultCodegenConfig.java:3773)
        at io.swagger.codegen.v3.generators.DefaultCodegenConfig.fromResponse(DefaultCodegenConfig.java:2192)
        at io.swagger.codegen.v3.generators.DefaultCodegenConfig.fromOperation(DefaultCodegenConfig.java:1903)
        at io.swagger.codegen.v3.DefaultGenerator.processOperation(DefaultGenerator.java:847)
        at io.swagger.codegen.v3.DefaultGenerator.processPaths(DefaultGenerator.java:774)
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:419)
        at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:722)
        at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:342)
        at java.lang.Thread.run(Thread.java:748)
Exception in thread "Thread-1" java.lang.RuntimeException: Could not process operation:
  Tag: class Tag {
    name: /authentication
    description: null
    externalDocs: null
}
  Operation: loginWithToken
  Resource: post /authentication/login_with_token
  Exception: null
        at io.swagger.codegen.v3.DefaultGenerator.processOperation(DefaultGenerator.java:872)
        at io.swagger.codegen.v3.DefaultGenerator.processPaths(DefaultGenerator.java:774)
        at io.swagger.codegen.v3.DefaultGenerator.generateApis(DefaultGenerator.java:419)
        at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:722)
        at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:342)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
        at io.swagger.codegen.v3.generators.DefaultCodegenConfig.getSchemaFromResponse(DefaultCodegenConfig.java:3773)
        at io.swagger.codegen.v3.generators.DefaultCodegenConfig.fromResponse(DefaultCodegenConfig.java:2192)
        at io.swagger.codegen.v3.generators.DefaultCodegenConfig.fromOperation(DefaultCodegenConfig.java:1903)
        at io.swagger.codegen.v3.DefaultGenerator.processOperation(DefaultGenerator.java:847)
        ... 5 more
```

--- a/src/main/java/io/swagger/codegen/v3/generators/DefaultCodegenConfig.java
+++ b/src/main/java/io/swagger/codegen/v3/generators/DefaultCodegenConfig.java
@@ -3770,7 +3770,9 @@ public abstract class DefaultCodegenConfig implements CodegenConfig {
         Schema schema = null;
         for (String contentType : response.getContent().keySet()) {
             schema = response.getContent().get(contentType).getSchema();
-            schema.addExtension("x-content-type", contentType);
+            if (schema != null) {
+                schema.addExtension("x-content-type", contentType);
+            }
             break;
         }
         return schema;

